wayback/_client.py

```diff
@@ -215,6 +215,17 @@ def cdx_hash(content):
     return b32encode(hashlib.sha1(content).digest()).decode()
 
 
+def read_and_close(response):
+    # Read content so it gets cached and close the response so
+    # we can release the connection for reuse. See:
+    # https://github.com/psf/requests/blob/eedd67462819f8dbf8c1c32e77f9070606605231/requests/sessions.py#L160-L163
+    try:
+        response.content
+    except (ChunkedEncodingError, ContentDecodingError, RuntimeError):
+        response.raw.read(decode_content=False)
+    response.close()
```

Comment:
Far be it from me to try to improve on an implementation copied straight out of `requests`, but I might move this inside a `finally:` block in case any other unexpected Exception types are raised above.

Reply:
Oh, nice thinking! 👍